app/controllers/application_controller.rb

```diff
@@ -31,4 +32,12 @@ def current_user_can?(*args)
   def render_http_status(status)
     render plain: Rack::Utils::HTTP_STATUS_CODES[status], status: status
   end
+
+  def set_current_user
+    Current.user = current_user
+    yield
+  ensure
+    # to address the thread variable leak issues in Puma/Thin webserver
```

Comment:
Which are? Could we include a link to documentation explaining this?

Reply:
basically threads being reused as far as I can tell.
https://stackoverflow.com/a/8291218/738371